feature_layers/display-subtype-feature-layer/src/main/java/com/esri/samples/feature_layers/DisplaySubtypeFeatureLayerController.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2019 Esri.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.esri.samples.feature_layers;
+
+import java.io.File;
+import java.util.Objects;
+import java.util.Scanner;
+
+import com.esri.arcgisruntime.arcgisservices.LabelDefinition;
+import com.esri.arcgisruntime.data.ServiceFeatureTable;
+import com.esri.arcgisruntime.geometry.Envelope;
+import com.esri.arcgisruntime.geometry.SpatialReferences;
+import com.esri.arcgisruntime.layers.SubtypeFeatureLayer;
+import com.esri.arcgisruntime.layers.SubtypeSublayer;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.symbology.Renderer;
+import com.esri.arcgisruntime.symbology.SimpleMarkerSymbol;
+import com.esri.arcgisruntime.symbology.SimpleRenderer;
+import com.esri.arcgisruntime.symbology.Symbol;
+import javafx.fxml.FXML;
+
+import com.esri.arcgisruntime.mapping.view.MapView;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.Label;
+import javafx.scene.control.ToggleButton;
+
+public class DisplaySubtypeFeatureLayerController {
+
+  @FXML private ToggleButton toggleRendererButton;
+  @FXML private MapView mapView;
+  @FXML private Label currentMapScaleLabel;
+  @FXML private Label minScaleLabel;
+  @FXML private CheckBox checkBox;
+  private SubtypeSublayer sublayer;
+  private Renderer originalRenderer;
+  
+  public void initialize() {
+
+    try {
+
+      // create a map with streets night vector basemap and add it to the map view
+      ArcGISMap map = new ArcGISMap();
+      map.setBasemap(Basemap.createStreetsNightVector());
+      mapView.setMap(map);
+      // display the current map scale 
+      mapView.addMapScaleChangedListener(mapScaleChangedEvent ->
+        currentMapScaleLabel.setText("Current Map Scale: 1:" + Math.round(mapView.getMapScale())));
+
+      // set the viewpoint to Naperville, Illinois
+      Viewpoint initialViewpoint = new Viewpoint(new Envelope(-9812691.11079696, 5128687.20710657,
+        -9812377.9447607, 5128865.36767282, SpatialReferences.getWebMercator()));
+      map.setInitialViewpoint(initialViewpoint);
+
+      // create a subtype feature layer from the service feature table, and add it to the map
+      final String serviceFeatureTableUrl = "https://sampleserver7.arcgisonline" +
```

Comment:
I suppose the String is only ever being used once, so you could pop it straight into the `new ServiceFeatureTable()` below?